### ISSUE TITLE
Build fails on openSUSE Tumbleweed system

### ISSUE THREAD

**Issue by patrikreiske** [JuliaIO/ImageMagick.jl#113]
Building the package fails on my openSUSE Tumbleweed system. 

```
julia> versioninfo()
Julia Version 0.6.2
Commit d386e40 (2017-12-13 18:08 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH Sandybridge)
  LAPACK: libopenblas_openmp.so.0
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, ivybridge)

julia> Pkg.build("ImageMagick")
INFO: Building FFTW
INFO: Building SpecialFunctions
INFO: Building ImageMagick
=====================================================================================[ ERROR: ImageMagick ]=====================================================================================

LoadError: None of the selected providers can install dependency libwand.
Use BinDeps.debug(package_name) to see available providers

while loading /home/patrik/.julia/v0.6/ImageMagick/deps/build.jl, in expression starting on line 117

================================================================================================================================================================================================

========================================================================================[ BUILD ERRORS ]========================================================================================

WARNING: ImageMagick had build errors.

 - packages with build errors remain installed in /home/patrik/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("ImageMagick")`
 - build a single package by running its `deps/build.jl` script

================================================================================================================================================================================================
```

I did some investigation on my own but didn't manage to fix the problem (started using Julia a few days ago).

On openSUSE Tumbleweed, the library is installed though the package "libGraphicsMagickWand-Q16-2" and to be found in the filesystem as /usr/lib64/libGraphicsMagickWand-Q16.so.2 but since the build does neither have and appropriate entry in the aliases vector nor package provider, the build fails I think.

Hope this helps fixing the problem!

**Comment by patrikreiske** [comment]
I did some further investigation and found out that actually the package `libMagickWand-7_Q16HDRI5` provides the library, as you see in the snipped below.

```
T430:~ # LANG="en" zypper info --provides libMagickWand-7_Q16HDRI5
Loading repository data...
Reading installed packages...


Information for package libMagickWand-7_Q16HDRI5:
-------------------------------------------------
Repository     : Haupt-Repository (OSS)           
Name           : libMagickWand-7_Q16HDRI5         
Version        : 7.0.7.25-1.1                     
Arch           : x86_64                           
Vendor         : openSUSE                         
Installed Size : 1.2 MiB                          
Installed      : Yes                              
Status         : up-to-date                       
Source package : ImageMagick-7.0.7.25-1.1.src     
Summary        : C runtime library for ImageMagick
Description    :                                  
    ImageMagick is a robust collection of tools and libraries to read,
    write, and manipulate an image in many image formats, including popular
    formats like TIFF, JPEG, PNG, PDF, PhotoCD, and GIF. With ImageMagick,
    you can create images dynamically, making it suitable for Web
    applications. You can also resize, rotate, sharpen, color-reduce, or
    add special effects to an image and save your completed work in many
    different image formats. Image processing operations are available from
    the command line as well as through C, C++, and Perl-based programming
    interfaces.
Provides       : [4]                              
    libMagickWand-7.Q16HDRI.so.5()(64bit)
    libMagickWand-7.Q16HDRI.so.5(VERS_5.0)(64bit)
    libMagickWand-7_Q16HDRI5 = 7.0.7.25-1.1
    libMagickWand-7_Q16HDRI5(x86-64) = 7.0.7.25-1.1

T430:~ # find /usr/lib64/ -name "libMagickWand*"
/usr/lib64/libMagickWand-7.Q16HDRI.so.5
/usr/lib64/libMagickWand-7.Q16HDRI.so.5.0.0
```

I tried fixing the issue on my own, editing the package's `build.jl` file like this:

```
33c33
< suffixes    = ["", "-Q16", "-6.Q16", "-Q8"]
---
> suffixes    = ["", "-Q16", "-6.Q16", "-Q8", "-7.Q16"]
55a56
>     provides(Zypper, "libMagickWand-7_Q16HDRI5", libwand)
```

But I didn't manage it to work:

```
julia> BinDeps.debug("ImageMagick")
INFO: Reading build script...
The package declares 1 dependencies.
 - Library "libwand"
    - Providers:
      - BinDeps.AptGet package libmagickwand4 (can't provide)
      - BinDeps.AptGet package libmagickwand5 (can't provide)
      - BinDeps.AptGet package libmagickwand-6.q16-2 (can't provide)
      - BinDeps.Pacman package imagemagick (can't provide)
      - BinDeps.Yum package ImageMagick (can't provide)
      - BinDeps.Zypper package libMagickWand-7_Q16HDRI5

julia> Pkg.build("ImageMagick")
INFO: Building FFTW
INFO: Building SpecialFunctions                                                                                
INFO: Building ImageMagick                                                                                     
============================================[ ERROR: ImageMagick ]=============================================

LoadError: Will not force zypper to rebuild dependency "libwand".
Please make any necessary adjustments manually (This might just be a version upgrade)                          
while loading /home/patrik/.julia/v0.6/ImageMagick/deps/build.jl, in expression starting on line 119

===============================================================================================================

===============================================[ BUILD ERRORS ]================================================

WARNING: ImageMagick had build errors.

 - packages with build errors remain installed in /home/patrik/.julia/v0.6                                     
 - build the package(s) and all dependencies with `Pkg.build("ImageMagick")`                                   
 - build a single package by running its `deps/build.jl` script                                                

===============================================================================================================
```

Sadly I don't have any experience in Julia, yet. And since several other packages (image processing, …) require this package, so I need you guys to help me out.

**Comment by SimonDanisch** [comment]
>LoadError: Will not force zypper to rebuild dependency "libwand".

This seems to be an error generated by zypper ... I don't even know zypper, so I guess you'll need the help of someone familiar with that package manager... Maybe uninstall imagemagick manually with zypper and see what happens?

**Comment by patrikreiske** [comment]
Since uninstalling it would also force me to uninstall several other packages, I'd like to avoid this test. Is there any way to get a more detailed error message?

**Comment by SimonDanisch** [comment]
You could open an issue with bindeps to see if whoever wrote this knows more: https://github.com/JuliaLang/BinDeps.jl/blob/master/src/dependencies.jl#L482
You might also want to locally go to this line and turn this into a warning or ignor it ;) 
It's weird that it's an error, while the message suggest that it's non fatal.

**Comment by SimonDanisch** [comment]
Might be fixed by #121! Would be nice if you can confirm!

**Comment by patrikreiske** [comment]
That's great news! Sadly, I don't have access to an openSUSE system at the moment. But I'll try to find some time to setup a VM and test it within the next week.

**Comment by patrikreiske** [comment]
Sorry for my late response! Won't be able to test this. At least not in the
next months.

Am Mi., 11. Juli 2018 um 21:44 Uhr schrieb Simon <notifications@github.com>:

**Comment by timholy** [comment]
Should be fixed if you use Julia 1.3+
